### PR TITLE
Error message when title already exists

### PR DESCRIPTION
**Pull request by denektenina**
# Description

Link to Jira issue: [NP-47079](https://unit.atlassian.net/browse/NP-47079)

When creating a new registration and entering a name the user should be notified if this exact name already exists in NVA.

The way it is today, nothing happens when we type in the title of an existing registration:

image

In this PR we get a warning under the search box so that we can choose to stop registering if it already exists:

image

**Features in this PR:**
* When typing the title a debounced query will search for publications with the same name. Since the search returns registrations with similar names, we are searching among them for one that fits exactly (difference in uppercase / lowercase is accepted)
* New spinner while searching
* Warning icon in title input field if there exists one with same name
* Custom error message displayed if there exists one with same name
* Click the title of the registration with the same name to open a window with the registration searched for

# Checklist

Ensure that the changes are aligned with the expectations of PO/designer before marking the PR as ready for review.

Also ensure that the following criterias are met:

- [x] The changes are working as expected
- [x] The changes are tested OK for different screen sizes
- [x] The changes are tested OK for a11y
- [x] Interactive elements have data-testids
- [x] I have done a QA of my own changes

_Note: some of these criterias may not always be relevant, and can simply be marked as completed._

[NP-47079]: https://unit.atlassian.net/browse/NP-47079?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by joachimjorgensen — DISMISSED** [review]
Looks good, and works as expected (?) 👍 (Question-mark relates to my comment)

File: src/pages/registration/DescriptionPanel.tsx [hunk 1]
```diff
@@ -35,13 +49,58 @@ export const DescriptionPanel = () => {
...
+            <Typography sx={{ fontWeight: 'bold' }}>{t('common.result')}</Typography>
+            <Link
+              target="_blank"
+              data-testid={dataTestId.registrationLandingPage.duplicateRegistrationSearchLink}
+              to={{
+                pathname: UrlPathTemplate.Home,
+                search: registrationWithSameName.entityDescription?.mainTitle
+                  ? `?query=${encodeURIComponent(registrationWithSameName.entityDescription.mainTitle)}`
+                  : '',
+              }}>
```
[the remark this answers is not in the document]
Comment (denektenina) [reply src/pages/registration/DescriptionPanel.tsx:92]: I asked Mayli about this and she said: "Det er ikke sikkert brukeren er registrert på resultatet som allerede er registrert. Så derfor så blir det åpnet en ny fanen på søk med artikkelen med det samme tittelen.", but she was unsure if the user should get into the registration if he or she was registered to it. I am also unsure if we should show all similar hits or just the exact one as I did. I can write these questions in the issue and make a new issue if JE wants me to change it.

[the remark this answers is not in the document]
Comment (joachimjorgensen) [reply src/pages/registration/DescriptionPanel.tsx]: Does it matter if the user is registered on the publication? I think all registrations are open to everyone?
Yes, merge this, and ask what JE wants :)

File: src/pages/registration/SameNameWarning.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        bgcolor: 'secondary.light',
+        width: '100%',
+        padding: '0.75rem',
+        display: 'flex',
+        flexDirection: 'column',
+        gap: '0.5rem',
+      }}>
+      <Box sx={{ bgcolor: 'primary.light', color: 'white', p: '0.5rem', borderRadius: '0.25rem' }}>
+        {t('registration.description.duplicate_title_warning')}
+      </Box>
```
Comment (joachimjorgensen) [inline src/pages/registration/SameNameWarning.tsx:26]: I'm all for the new component `<SameNameWarning ... />`, but I was thinking of a `<Notification title="" description="" type="info|warning" />` component just for this blue box. Can't say I have seen a notification like this elsewhere, so it's okay to leave here for now. But if there are more notifications like this in the future we should probably extract it into a component.

**Review by joachimjorgensen — APPROVED** [review]

File: src/pages/registration/DescriptionPanel.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (denektenina) [reply src/pages/registration/DescriptionPanel.tsx]: Yes I thought that was strange to.

File: src/pages/registration/SameNameWarning.tsx (hunk 2 above)
Comment (denektenina) [reply src/pages/registration/SameNameWarning.tsx:26]: Good point!

**Pull request merged**